realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/util/KotlinTestUtils.kt

```diff
@@ -29,3 +31,21 @@ inline fun <reified T> ErrorCollector.assertFailsWith(block : () -> Unit){
         }
     }
 }
+
+/**
+ * A **resource container** to keep references for objects that should later be closed.
+ */
+class ResourceContainer : Closeable {
```

Comment:
As discussed in Slack, we should probably see how far `realm.use { .. }` will get us

Reply:
`use` is useful where closing follows scope. But this was introduced to assist the tests where it does not follow scope. The specific problematic test are those where we need to close the Realm in a callback passed to the config. In some of the test we also asserts stuff after closing, so we cannot use the collecting mechanism on the test looper to just close them after completion. 

We should consider always adding the Realm as an argument to the callback, so that it would be way easier to access and reuse the callbacks across multiple configs. 
Another